### PR TITLE
Detect Dart pub packages and their dependencies from pubspec.yaml

### PR DESCRIPTION
**Pull request by mrmayurgithub**
Fixes #0000

### Tasks

* [ ] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [ ] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: src/packagedcode/pubspec.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    default_web_baseurl = 'https://dart.dev/tools/pub/pubspec
+    default_download_baseurl = 'https://dart.dev/tools/pub/pubspec'
+    default_api_baseurl = 'https://dart.dev/tools/pub/pubspec'
+
+    @classmethod
+    def recognize(cls, location):
+        yield parse(location)
+
+    def repository_homepage_url(self, baseurl=default_web_baseurl):
+        return '{}/pods/{}'.format(baseurl, self.name)
```
Comment (rpotter12) [inline src/packagedcode/pubspec.py:70]: I think you have used Cocoapods link here. Update this with dart package.

File: src/packagedcode/pubspec.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    def repository_download_url(self):
+        return '{}/archive/{}.zip'.format(self.homepage_url, self.version)
```
Comment (rpotter12) [inline src/packagedcode/pubspec.py:73]: Check this also once again. You might be using pods link.

File: src/packagedcode/pubspec.py (hunk 1 above)
Comment (rpotter12) [inline src/packagedcode/pubspec.py:70]: I think you have used Cocoapods link here. Update this with dart package.

File: src/packagedcode/pubspec.py (hunk 2 above)
Comment (rpotter12) [inline src/packagedcode/pubspec.py:73]: Check this also once again. You might be using pods link.

File: src/packagedcode/pubspec.py (hunk 1 above)
Comment (mrmayurgithub) [reply src/packagedcode/pubspec.py:70]: > 
> 
> I think you have used Cocoapods link here. Update this with dart package.

Ok, I'll do that.

File: src/packagedcode/pubspec.py (hunk 2 above)
Comment (mrmayurgithub) [reply src/packagedcode/pubspec.py:73]: Actually, I was taking help from many other packagedcode files :P. I'll correct it. :)

**Comment by mrmayurgithub** [comment]
> 
> 
> @mrmayurgithub This is looking good. You now need to link this with the scancode software. Please check #2097 PR. Check how we have link these files with the software and how we have written tests for this?

Ok

**Comment by pombredanne** [comment]
@mrmayurgithub ping :) I would would to not see finished and merged ... what's left to do on your side?

**Comment by mrmayurgithub** [comment]
> 
> 
> @mrmayurgithub ping :) I would would to not see finished and merged ... what's left to do on your side?

@pombredanne I am not getting how to write the tests. :) I have never written tests.

**Comment by pombredanne** [comment]
@mrmayurgithub you could use the examples of other tests we have as a base

**Comment by pombredanne** [comment]
ping?

**Comment by mrmayurgithub** [comment]
> 
> 
> ping?

I am sorry  to say, but I literally don't know how to write tests, and I am unable to understand through the files too. :(

**Comment by pombredanne** [comment]
@mrmayurgithub I would like to merge this but I can't without some tests... can you point to some URLs of pubspec files that could be used for testing?

**Comment by pombredanne** [comment]
@mrmayurgithub I ended cherry picking, updating and using this https://github.com/nexB/scancode-toolkit/pull/2571 ...
So I am closing this now.

**Pull request closed without merging**